### PR TITLE
D8CORE-1943 Fix the link widget and how it handles the suggestions and absolute urls

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Fixed wrong logic when user enters an absolute ur.

# Need Review By (Date)
- 5/11

# Urgency
- High

# Steps to Test
1. Checkout this branch
1. clear browser cache
1. create a page with a card
1. enter a variety of urls, absolute, relative, type a page name and click the suggestion.
1. validate it all saves accurately

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: js/src/Components/Widgets/LinkWidget.js [hunk 1]
```diff
@@ -16,18 +16,22 @@ export const LinkWidget = ({fieldId, defaultValue, onFieldChange, settings}) =>
...
-        uri: getUriFromString(newUri)
-      }]);
-
-      if (newUri.substr(0, 1) === '/' || newUri.substr(0, 1) === '<') {
+      // If the user enters a url that starts with some characters, we dont want to fetch suggestions that will just
+      // be empty anyways. This includes absolute urls, <front> and relative urls.
+      if (
+        newUri.substr(0, 1) === '/' ||
+        newUri.substr(0, 1) === '<' ||
+        newUri.substr(0, 4) === 'http'
```
Comment (sherakama) [inline js/src/Components/Widgets/LinkWidget.js]: What about other protocols? ftp, smb, or streaming URLs? Perhaps validate for the pattern `[a-z0-9]*:\/\/`

Comment (pookmish) [reply js/src/Components/Widgets/LinkWidget.js]: do people put those in there?!

Comment (sherakama) [reply js/src/Components/Widgets/LinkWidget.js]: Someone somewhere has a use case for it and it wouldn't be too much for us to allow it here.

Comment (pookmish) [reply js/src/Components/Widgets/LinkWidget.js]: well its not that it "doesnt allow" this just avoids unnecessary suggestion api calls. but i get what you're saying. no need to provide suggestions for ftp urls

Comment (pookmish) [reply js/src/Components/Widgets/LinkWidget.js]: I swapped with that protocol regex check

**Review by sherakama — APPROVED** [review]

**Pull request merged**